amtl/am-hashtable.h

```diff
@@ -319,10 +319,18 @@ class HashTable : private AllocPolicy
     Probulator probulator(hash, capacity_);
 
     Entry *e = &table_[probulator.entry()];
+    Entry *firstRemoved = nullptr;
     for (;;) {
-      if (!e->isLive())
+      if (e->isFree()) {
+        if (firstRemoved)
+         e = firstRemoved;
         break;
-      if (e->sameHash(hash) && HashPolicy::matches(key, e->payload()))
+      }
+      if (e->removed()) {
+        if (!firstRemoved)
+         firstRemoved = e;
+      }
+      else if (e->sameHash(hash) && HashPolicy::matches(key, e->payload()))
```

Comment:
I thought that too, but then you're checking e->removed twice, or you're actually checking the entire hash. Stylistically bad, but it's better generated code.